src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/WorkspaceConfigurationChangedListener.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.AspNetCore.Razor.LanguageServer;
+using Microsoft.CodeAnalysis.Razor.Editor;
+using Microsoft.VisualStudio.Editor.Razor;
+using Microsoft.VisualStudio.LanguageServer.ContainedLanguage;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Newtonsoft.Json.Linq;
+using DidChangeConfigurationParams = OmniSharp.Extensions.LanguageServer.Protocol.Models.DidChangeConfigurationParams;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor
+{
+    [Shared]
+    [Export(typeof(EditorSettingsChangedTrigger))]
+    internal class WorkspaceConfigurationChangedListener : EditorSettingsChangedTrigger
+    {
+        private readonly LSPRequestInvoker _requestInvoker;
+
+        [ImportingConstructor]
+        public WorkspaceConfigurationChangedListener(LSPRequestInvoker requestInvoker!!)
+        {
+            _requestInvoker = requestInvoker;
+        }
+
+        public override void Initialize(EditorSettingsManager editorSettingsManager)
+        {
+            editorSettingsManager.Changed += EditorSettingsManager_Changed;
+        }
+
+        private void EditorSettingsManager_Changed(object sender, EditorSettingsChangedEventArgs args)
+        {
+            _ = EditorSettingsManager_ChangedAsync();
```

Comment:
Why implement this with an event? You have an async method that is fired-and-forgetted to fire the event, that does so on the UI thread, and then triggers that listen for the event and fire-and-forget their method that does stuff.

Why not just have this override a method on the base class, and have the manager spin through its triggers and trigger them?